core/src/splitting/RegressionSplittingRule.cpp

```diff
@@ -64,6 +64,9 @@ bool RegressionSplittingRule::find_best_split(const Data& data,
   // For all possible split variables
   for (auto& var : possible_split_vars) {
     // Use faster method for both cases
+    // (tradeoff between nlog(n) and N)
```

Comment:
Maybe we could move this to the method comment on `find_best_split` in `RegressionSplittingRule.h` and expand it? In particular we could define `n` and `N` and give some background about how this heuristic came about. Right now I think it might be too concise to help a newcomer.